.github/workflows/dependency-review-on-pr.yml

```diff
@@ -0,0 +1,23 @@
+name: Dependency review for pull requests
+
+on:
+  pull_request:
+
+permissions:
+  contents: write
+
+jobs:
+  dependency-submission:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-java@v4
+        with:
+          distribution: temurin
+          java-version: 17
+
+      - name: Generate and submit dependency graph
```

Comment:
@whi-tw this will be rejected from the PR branch as it doesn't collect metadata so this action can be removed except on push to main.